src/Cache.php

```diff
@@ -0,0 +1,59 @@
+<?php
```

Comment:
Maybe [PSR6](https://github.com/php-fig/cache) or [PSR16](https://github.com/php-fig/simple-cache) could be considered instead of a concrete class

Reply:
Considering it stays internal and is strictly used as an in memory/memoization cache I’d say PSR16 is probably more than we need and doesn’t provide any significant trade offs in this instance.